apps/settings/test/unit/panels/homescreens/homescreen_name_test.js

```diff
@@ -0,0 +1,74 @@
+/* global MockNavigatorSettings */
+'use strict';
+
+require('/shared/test/unit/mocks/mock_navigator_moz_settings.js');
+
+suite('homescreens > homescreen name', () => {
+  var modules = [
+    'shared_mocks/mock_settings_listener',
+    'unit/mock_apps_cache',
+    'panels/homescreens/homescreen_name'
+  ];
+
+  var map = {
+    '*': {
+      'shared/settings_listener': 'shared_mocks/mock_settings_listener',
+      'modules/apps_cache': 'unit/mock_apps_cache'
+    }
+  };
+
+  var homescreenName;
+
+  var realNavigatorSettings;
+  var realSettingsListener;
+
+  var mockSettingsListener;
+  var mockAppsCache;
+
+  suiteSetup(() => {
+    realNavigatorSettings = navigator.mozSettings;
+    navigator.mozSettings = MockNavigatorSettings;
+  });
+
+  suiteTeardown(() => {
+    navigator.mozSettings = realNavigatorSettings;
+    realNavigatorSettings = null;
+
+    window.SettingsListener = realSettingsListener;
+  });
+
+  setup(done => {
+    var requireCtx = testRequire([], map, () => {
+    });
+    requireCtx(modules, (MockSettingsListener, MockAppsCache,
+                         HomescreenName) => {
+      homescreenName = HomescreenName();
+      realSettingsListener = window.SettingsListener;
+      window.SettingsListener = MockSettingsListener;
+      mockSettingsListener = MockSettingsListener;
```

Comment:
actually you can use `window.SettingsListener` to do any thing now, not required extra `mockSettingsListener`